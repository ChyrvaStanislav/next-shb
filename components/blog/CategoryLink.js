import React from 'react';
import PropTypes from 'prop-types';
import { Link, Typography } from '@/components/html';
import classNames from 'classnames';
import styles from './CategoryLink.module.scss';

const CategoryLink = ({
  category, setCategory, style, setMeta, querySearch, isActive
}) => {
  if (isActive) {
    return (
      <Typography
        variant="p_links"
        className={classNames(style, styles.link2, `topic${category?.label?.replace(/ /g, '')}`)}
        key={`${category?.id}-link`}
      >
        {category?.label}
      </Typography>
    );
  }

  const categoryAdditionalParam = `&query=${querySearch}`;

  return (
    <Link
      href={`/${category?.id}${querySearch ? categoryAdditionalParam : ''}`}
      onClick={() => {
        setCategory(category?.id);
        if (category?.meta) {
          setMeta(category.meta);
        }
      }}
      className={classNames(style, styles.link2, `topic${category?.label?.replace(/ /g, '')}`)}
      key={`${category?.id}-link`}
      type="link"
      data-testid="category-link"
    >
      {category?.label}
    </Link>
  );
};
CategoryLink.propTypes = {
  category: PropTypes.shape({
    id: PropTypes.string,
    label: PropTypes.string,
  }),
  setCategory: PropTypes.func,
  style: PropTypes.string.isRequired,
  setMeta: PropTypes.func.isRequired,
  querySearch: PropTypes.string,
  isActive: PropTypes.bool,
};

CategoryLink.defaultProps = {
  category: {
    id: '',
    label: '',
  },
  setCategory: () => {},
  querySearch: null,
  isActive: false,
};

export default CategoryLink;
