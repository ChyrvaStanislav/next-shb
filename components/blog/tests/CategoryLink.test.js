/* eslint-disable react/style-prop-object */

import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import CategoryLink from '../CategoryLink';

const category = {
  meta: {
    title: 'Data Analytics Articles | EPAM SolutionsHub',
    description: 'Read useful articles about Data Analytics.'
  },
  label: 'Analytics',
  id: 'analytics',
};

const setCategory = jest.fn();
const setMeta = jest.fn();

beforeEach(() => {
  jest.clearAllMocks();
});

const Component = (props) => (
  <BrowserRouter>
    <CategoryLink style="some-styles" setMeta={setMeta} setCategory={setCategory} category={category} {...props} />
  </BrowserRouter>
);

describe('CategoryLink snapshots', () => {
  test('should NOT active CategoryLink match snapshot', () => {
    const { asFragment } = render(<Component />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('should active CategoryLink match snapshot', () => {
    const { asFragment } = render(<Component isActive={true} />);

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('CategoryLink functions work', () => {
  test('should setMeta triggered with meta data', () => {
    render(<Component />);
    fireEvent.click(screen.getByTestId('category-link'));

    expect(setMeta).toBeCalledWith({
      title: 'Data Analytics Articles | EPAM SolutionsHub',
      description: 'Read useful articles about Data Analytics.'
    });
  });

  test('should setCategory triggered with id', () => {
    render(<Component />);
    fireEvent.click(screen.getByTestId('category-link'));

    expect(setCategory).toBeCalledWith('analytics');
  });
});
