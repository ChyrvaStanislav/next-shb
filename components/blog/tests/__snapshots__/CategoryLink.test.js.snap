// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryLink snapshots should NOT active CategoryLink match snapshot 1`] = `
<DocumentFragment>
  <a
    class="p_links some-styles link2 topicAnalytics"
    data-testid="category-link"
    href="/blog?category=analytics"
    other=""
  >
    Analytics
  </a>
</DocumentFragment>
`;

exports[`CategoryLink snapshots should active CategoryLink match snapshot 1`] = `
<DocumentFragment>
  <p
    class="p_links some-styles link2 topicAnalytics"
    other=""
  >
    Analytics
  </p>
</DocumentFragment>
`;
